src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/PartiallyLoweredLocalFunctionReference.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// This represents a partially lowered local function reference (e.g.,
+    /// a local function call or delegate conversion) with relevant proxies
+    /// attached. It will later be rewritten by the
+    /// <see cref="LambdaRewriter.LocalFunctionReferenceRewriter"/> into a
+    /// proper call.
+    /// </summary>
+    internal class PartiallyLoweredLocalFunctionReference : BoundExpression
```

Comment:
Should this class be defined in `BoundNodes.xml`? Doing so would add it to `BoundTreeVisitor`, avoiding the need for special case in `LocalFunctionReferenceRewriter.Visit`. It would also add it to `BoundKind`. (We're currently using `underlying.Kind` here. Is there a chance that visitors will encounter this node and (incorrectly) cast it to the type of `underlying`?)